### PR TITLE
Use Collection expressions for all language reference samples

### PR DESCRIPTION
**Pull request by BillWagner**
Contributes to #37000 

Update all the samples in the C# Language reference to use collection expressions where applicable.

I converted a number of the source files to use file scoped namespaces. Hide whitespace to better see actual diffs.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/language-reference/operators/stackalloc.md](https://github.com/dotnet/docs/blob/7a9861a616d29ddd2cf4ef9dd9881074890c0e74/docs/csharp/language-reference/operators/stackalloc.md) | [stackalloc expression (C# reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/stackalloc?branch=pr-en-us-37993) |

### CODE REVIEW

**Comment by BillWagner** [comment]
closing and reopening for a fresh build

File: docs/csharp/language-reference/builtin-types/snippets/shared/VoidType.cs [hunk 1]
```diff
@@ -4,7 +4,7 @@ public static class VoidType
 {
     public static void Examples()
     {
-        Display(new[] { 1, 2, 9 });
+        Display([ 1, 2, 9 ]);
```
Comment (IEvangelist) [inline docs/csharp/language-reference/builtin-types/snippets/shared/VoidType.cs:7]:
```suggestion
        Display([1, 2, 9]);
```

File: docs/csharp/language-reference/operators/snippets/shared/Overview.cs [hunk 2]
```diff
@@ -23,8 +25,8 @@ private static void GeneralExamples()
         
         string s = "String literal";
         char l = s[s.Length - 1];
-        
-        var numbers = new List<int>(new[] { 1, 2, 3 });
+
+        List<int> numbers = [..collection];
```
Comment (IEvangelist) [inline docs/csharp/language-reference/operators/snippets/shared/Overview.cs:29]: This is different than the original, is that intentional?

File: docs/csharp/language-reference/operators/snippets/shared/TypeTestingAndConversionOperators.cs [hunk 3]
```diff
@@ -76,7 +77,7 @@ private static void IsOperatorDeclarationPattern()
     private static void AsOperator()
     {
         // <SnippetAsOperator>
-        IEnumerable<int> numbers = new[] { 10, 20, 30 };
+        IEnumerable<int> numbers = [ 10, 20, 30 ];
```
Comment (IEvangelist) [inline docs/csharp/language-reference/operators/snippets/shared/TypeTestingAndConversionOperators.cs:80]:
```suggestion
        IEnumerable<int> numbers = [10, 20, 30];
```

File: samples/snippets/csharp/concepts/methods/swap107.cs [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ﻿// <Snippet107>
-using System;
 
+using System;
```
Comment (IEvangelist) [inline samples/snippets/csharp/concepts/methods/swap107.cs:3]:
```suggestion
using System;

```

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**